File: Ejercicios Conexion/C++/MongoDB.cpp
#include <iostream>
#include <mongocxx/client.hpp>
#include <mongocxx/instance.hpp>
#include <mongocxx/uri.hpp>
#include <bsoncxx/builder/stream/document.hpp>

int main() {
    // Iniciar MongoDB
    mongocxx::instance instancia{};

    try {
        // Conectar a MongoDB
        mongocxx::client cliente{mongocxx::uri{"mongodb://localhost:27017/"}};
        std::cout << "✅ Conexión exitosa a MongoDB" << std::endl;

        // Seleccionar la base de datos y la colección
        auto db = cliente["ecommerce3066478"];
        auto coleccion = db["Clientes1"];

        // Crear documento
        bsoncxx::builder::stream::document doc{};
        doc << "Nombre" << "Carlos Martínez"
            << "Correo" << "carlos.martinez@email.com"
            << "Telefono" << "3204567890"
            << "Direccion" << "Carrera 12 #34-56"
            << "Creado desde" << "C++";

        // Insertar documento
        auto resultado = coleccion.insert_one(doc.view());
        if (resultado) {
            std::cout << "📌 Documento insertado con ID: " 
                      << resultado->inserted_id().get_oid().value.to_string() 
                      << std::endl;
        }

    } catch (const std::exception& e) {
        std::cerr << "❌ Error: " << e.what() << std::endl;
    }

    return 0;
}

File: Ejercicios Conexion/Java/src/main/java/MongoDB.java
import com.mongodb.client.*;
import org.bson.Document;

public class MongoDB {
    public static void main(String[] args) {
        try (MongoClient mongoClient = MongoClients.create("mongodb://localhost:27017")) {
            MongoDatabase database = mongoClient.getDatabase("ecommerce3066478");
            MongoCollection<Document> collection = database.getCollection("Clientes1");

            Document documento = new Document()
                    .append("Nombre", "Carlos")
                    .append("Correo", "carlos123@example.com")
                    .append("Telefono", "3012345678")
                    .append("Direccion", "Calle 10 #20-30")
                    .append("Creado_desde", "Java");

            collection.insertOne(documento);
            System.out.println("✅ Documento insertado con éxito.");
        } catch (Exception e) {
            System.out.println("❌ Error: " + e.getMessage());
        }
    }
}
